async-opcua-client/src/transport/connect.rs

```diff
@@ -26,8 +26,50 @@ pub trait Connector: Send + Sync {
         channel: Arc<RwLock<SecureChannel>>,
         outgoing_recv: tokio::sync::mpsc::Receiver<OutgoingMessage>,
         config: TransportConfiguration,
-        endpoint_url: &str,
     ) -> Result<TcpTransport, StatusCode>;
+
+    /// Get the default endpoint for this connector.
+    fn default_endpoint(&self) -> EndpointDescription;
+}
+
+/// Trait for types that can be used to create a connector.
+/// Implemented for `String`, `&str`, `&String`, and any type that implements the `Connector` trait.
+pub trait ConnectorBuilder: Send + Sync {
+    /// Create a new connector for the specific endpoint URL.
+    fn build(self) -> Result<Arc<dyn Connector + Send + Sync>, Error>;
```

Comment:
Maybe I just didn't look close enough at the PR, but why do we need the `Arc` here?